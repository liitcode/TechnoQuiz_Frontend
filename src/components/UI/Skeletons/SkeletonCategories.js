/* eslint-disable react/self-closing-comp */
/* eslint-disable react/no-array-index-key */
/* eslint-disable arrow-body-style */
import React from 'react';
import SkeletonElement from './SkeletonElement';
import Shimmer from './shimer';
import './skeleton.scss';

const SkeletonCategories = () => {
  const n = 11;
 return(
     <>
       {[...Array(n)].map(()=>(
       <div className="skeleton-card">
       <SkeletonElement type="circle" >
       <Shimmer />
       </SkeletonElement>
       <SkeletonElement type="rectangle" > <Shimmer /></SkeletonElement>
       
       </div>
       ))}
     </>
 )

}

export default SkeletonCategories;
