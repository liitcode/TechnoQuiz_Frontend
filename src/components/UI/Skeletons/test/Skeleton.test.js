import React from "react";
import renderer from "react-test-renderer";
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import storeConfig from '``../../../Redux/store/configureStore';
import SkeletonCategories from "../SkeletonCategories";

it('renders the component', () => {
    const tree = renderer.create(<Provider store={storeConfig}>
        <MemoryRouter><SkeletonCategories /></MemoryRouter></Provider>).toJSON();
    expect(tree).toMatchSnapshot();
})