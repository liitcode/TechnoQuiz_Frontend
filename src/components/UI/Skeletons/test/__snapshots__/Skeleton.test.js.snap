// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the component 1`] = `
Array [
  <div
    className="skeleton-card"
  >
    <div
      className="skeleton-circle"
    >
      <div
        className="shimmer-wrapper"
      >
        <div
          className="shimmer"
        />
      </div>
    </div>
    <div
      className="skeleton-rectangle"
    >
      <div
        className="shimmer-wrapper"
      >
        <div
          className="shimmer"
        />
      </div>
    </div>
  </div>,
  <div
    className="skeleton-card"
  >
    <div
      className="skeleton-circle"
    >
      <div
        className="shimmer-wrapper"
      >
        <div
          className="shimmer"
        />
      </div>
    </div>
    <div
      className="skeleton-rectangle"
    >
      <div
        className="shimmer-wrapper"
      >
        <div
          className="shimmer"
        />
      </div>
    </div>
  </div>,
  <div
    className="skeleton-card"
  >
    <div
      className="skeleton-circle"
    >
      <div
        className="shimmer-wrapper"
      >
        <div
          className="shimmer"
        />
      </div>
    </div>
    <div
      className="skeleton-rectangle"
    >
      <div
        className="shimmer-wrapper"
      >
        <div
          className="shimmer"
        />
      </div>
    </div>
  </div>,
  <div
    className="skeleton-card"
  >
    <div
      className="skeleton-circle"
    >
      <div
        className="shimmer-wrapper"
      >
        <div
          className="shimmer"
        />
      </div>
    </div>
    <div
      className="skeleton-rectangle"
    >
      <div
        className="shimmer-wrapper"
      >
        <div
          className="shimmer"
        />
      </div>
    </div>
  </div>,
  <div
    className="skeleton-card"
  >
    <div
      className="skeleton-circle"
    >
      <div
        className="shimmer-wrapper"
      >
        <div
          className="shimmer"
        />
      </div>
    </div>
    <div
      className="skeleton-rectangle"
    >
      <div
        className="shimmer-wrapper"
      >
        <div
          className="shimmer"
        />
      </div>
    </div>
  </div>,
  <div
    className="skeleton-card"
  >
    <div
      className="skeleton-circle"
    >
      <div
        className="shimmer-wrapper"
      >
        <div
          className="shimmer"
        />
      </div>
    </div>
    <div
      className="skeleton-rectangle"
    >
      <div
        className="shimmer-wrapper"
      >
        <div
          className="shimmer"
        />
      </div>
    </div>
  </div>,
  <div
    className="skeleton-card"
  >
    <div
      className="skeleton-circle"
    >
      <div
        className="shimmer-wrapper"
      >
        <div
          className="shimmer"
        />
      </div>
    </div>
    <div
      className="skeleton-rectangle"
    >
      <div
        className="shimmer-wrapper"
      >
        <div
          className="shimmer"
        />
      </div>
    </div>
  </div>,
  <div
    className="skeleton-card"
  >
    <div
      className="skeleton-circle"
    >
      <div
        className="shimmer-wrapper"
      >
        <div
          className="shimmer"
        />
      </div>
    </div>
    <div
      className="skeleton-rectangle"
    >
      <div
        className="shimmer-wrapper"
      >
        <div
          className="shimmer"
        />
      </div>
    </div>
  </div>,
  <div
    className="skeleton-card"
  >
    <div
      className="skeleton-circle"
    >
      <div
        className="shimmer-wrapper"
      >
        <div
          className="shimmer"
        />
      </div>
    </div>
    <div
      className="skeleton-rectangle"
    >
      <div
        className="shimmer-wrapper"
      >
        <div
          className="shimmer"
        />
      </div>
    </div>
  </div>,
  <div
    className="skeleton-card"
  >
    <div
      className="skeleton-circle"
    >
      <div
        className="shimmer-wrapper"
      >
        <div
          className="shimmer"
        />
      </div>
    </div>
    <div
      className="skeleton-rectangle"
    >
      <div
        className="shimmer-wrapper"
      >
        <div
          className="shimmer"
        />
      </div>
    </div>
  </div>,
  <div
    className="skeleton-card"
  >
    <div
      className="skeleton-circle"
    >
      <div
        className="shimmer-wrapper"
      >
        <div
          className="shimmer"
        />
      </div>
    </div>
    <div
      className="skeleton-rectangle"
    >
      <div
        className="shimmer-wrapper"
      >
        <div
          className="shimmer"
        />
      </div>
    </div>
  </div>,
]
`;
